ydb/core/tx/schemeshard/schemeshard__conditional_erase.cpp

```diff
@@ -141,7 +141,21 @@ struct TSchemeShard::TTxRunConditionalErase: public TSchemeShard::TRwTxBase {
         }
 
         const auto& settings = tableInfo->TTLSettings().GetEnabled();
-        const TDuration expireAfter = TDuration::Seconds(settings.GetExpireAfterSeconds());
+
+        TDuration expireAfter;
+         if (settings.TiersSize()) {
+            if (settings.TiersSize() > 1 || !settings.GetTiers(0).HasDelete()) {
+                LOG_WARN_S(ctx, NKikimrServices::FLAT_TX_SCHEMESHARD, "Invalid ttl setting: external storage tiers"
+                    << ": shardIdx: " << tableShardInfo.ShardIdx
+                    << ": pathId: " << shardInfo.PathId
+                    << ", at schemeshard: " << Self->TabletID());
+                return false;
+            }
+            expireAfter = TDuration::Seconds(settings.GetTiers(0).GetApplyAfterSeconds());
+         } else {
+            // legacy format
+            expireAfter = TDuration::Seconds(settings.GetExpireAfterSeconds());
+         }
```

Comment:
I think that code for getting `ExpireAfterSeconds` from ttl settings here and in the `schemeshard_validate_ttl.cpp:ValidateTtlSettings()` should be made into a single utility method.

Either that or those 2 places should be marked with comments pointing to each other.
But a single method is better.